### PR TITLE
Improvements of module PvAPI for branch 2.4

### PR DESCRIPTION
**Pull request by polarfuchs99**
Recently I have made the pull request #3804 for the branch "master" concerning improvements of the module PvAPI. This pull request has been accepted and merged. Steven Puttermans asked me to brinng the same changes and improvements also onto the branch 2.4, which I am doing now in this pull reques. If you want to know what I have changed in the PvAPI module, please see my comments in the pull request #3804 on the "master" branch.

The main changes of the PvAPI are in the file
cap_pvapi.cpp
This file is identical with that on the "master" branch with one exceptiion: on "master" the function getProperty is const where on the branch "2.4" it is not. Related defines have been added in the header file
highgui_c.h

Furthermore, I have removed an orthoigraphic error in one of the defines. I have replaced
CAP_PROP_MONOCROME
by
CAP_PROP_MONOCHROME
in all files which use this define. These are the files:
highgui_c.h
cap_dshow.cpp
gen_java.py

Sincerely,  Prof. Dr. Rudolf Haussmann

### CODE REVIEW

**Comment by StevenPuttemans** [comment]
:+1: it seems the error generated is not related to your PR but to a problem inside the buildbot! Simply be patient until it all builds fine!

**Comment by vpisarev** [comment]
@polarfuchs99, thanks for the PR. In the case of 2.4 branch we try to use very conservative approach and break as few things as possible. Thus, probably it makes sense to keep the enumeration constant CV_CAP_PROP_MONOCROME (in addition to adding its correctly spelled synonym CV_CAP_PROP_MONOCHROME)

File: modules/highgui/include/opencv2/highgui/highgui_c.h [hunk 1]
```diff
@@ -365,7 +365,7 @@ enum
     CV_CAP_PROP_CONVERT_RGB   =16,
     CV_CAP_PROP_WHITE_BALANCE_U =17,
     CV_CAP_PROP_RECTIFICATION =18,
-    CV_CAP_PROP_MONOCROME     =19,
```
Comment (StevenPuttemans) [inline modules/highgui/include/opencv2/highgui/highgui_c.h]: `CV_CAP_PROP_MONOCROME = CV_CAP_PROP_MONOCHROME`

We should add this to comply with the remark of @vpisarev

File: modules/highgui/src/cap_dshow.cpp [hunk 2]
```diff
@@ -2246,7 +2246,7 @@ int videoInput::getVideoPropertyFromCV(int cv_property){
         case CV_CAP_PROP_GAMMA:
             return VideoProcAmp_Gamma;
 
-        case CV_CAP_PROP_MONOCROME:
+        case CV_CAP_PROP_MONOCHROME:
```
Comment (StevenPuttemans) [inline modules/highgui/src/cap_dshow.cpp]: But then again, we will need double cases everywhere. @vpisarev is this really necessary?

**Comment by polarfuchs99** [comment]
@vpisarev  I agree with your consevative approach. Thus we should let CV_CAP_PROP_MONOCROME in the original form, even when it is misspelled. Then only the files cap_pvapi.cpp and highgui_c.h have changes for the pull request, and you can forget about the other files. Shall I make the undoing changes for MONOCROME or will you do this? I would agree, if you would undo the MONOCROME change and merge the stuff into the branch 2.4.

**Comment by StevenPuttemans** [comment]
@Rudi, I think you should simply provide both!

**Comment by polarfuchs99** [comment]
@vpisarev Once again, the changes to undo the MONOCROME orthographic correction is simply the following:
1. Leave my new file cap_pvapi.cpp unchanged.
2. In my new file highgui_c.h change the enumeration constant CV_CAP_PROP_MONOCHROME to its original spelling CV_CAP_PROP_MONOCROME.
3. Forget about my other files cap_dshow.cpp and gen_java.py.

**Comment by polarfuchs99** [comment]
@vpisarev I have made the changes to undo the orthographic correction for the enumeration constant CV_CAP_PROP_MONOCROME. Now there should be only the files 
cap_pvapi.cpp
highgui_c.h
to be taken into account in the merge.

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**